app/src/src_nonfreenet/org/breezyweather/sources/namem/NamemResultConverter.kt

```diff
@@ -0,0 +1,311 @@
+                        ),
+                        precipitationProbability = PrecipitationProbability(
+                            total = forecast.wwNPer
+                        ),
+                        wind = Wind(
+                            speed = forecast.wndN
+                        )
+                    )
+                )
+            )
+        }
+        dailyList.add(
+            Daily(
+                date = date,
+                day = HalfDay(
+                    weatherText = getWeatherText(context, forecast.wwD),
+                    weatherCode = getWeatherCode(forecast.wwD),
+                    temperature = Temperature(
+                        temperature = forecast.temD,
+                        apparentTemperature = forecast.temDFeel
+                    ),
+                    precipitationProbability = PrecipitationProbability(
+                        total = forecast.wwDPer
+                    ),
+                    wind = Wind(
+                        speed = forecast.wndD
+                    )
+                ),
+                night = dailyResult.fore5Day.getOrNull(i + 1)?.let {
+                    HalfDay(
+                        weatherText = getWeatherText(context, it.wwN),
+                        weatherCode = getWeatherCode(it.wwN),
+                        temperature = Temperature(
+                            temperature = it.temN,
+                            apparentTemperature = it.temNFeel
+                        ),
+                        precipitationProbability = PrecipitationProbability(
+                            total = it.wwNPer
+                        ),
+                        wind = Wind(
+                            speed = it.wndN
+                        )
+                    )
+                }
+            )
+        )
+    }
+    return dailyList
+}
+
+private fun getHourlyForecast(
+    hourlyResult: NamemHourlyResult,
+): List<HourlyWrapper> {
+    val hourlyList = mutableListOf<HourlyWrapper>()
+    hourlyResult.fore3Hours?.forEach {
+        if (it.fdate !== null) {
+            hourlyList.add(
+                HourlyWrapper(
+                    date = it.fdate,
+                    temperature = Temperature(
+                        temperature = it.tem?.toDoubleOrNull()
+                    ),
+                    precipitation = Precipitation(
+                        total = it.pre?.toDoubleOrNull()
+                    ),
+                    precipitationProbability = PrecipitationProbability(
+                        total = it.preProb?.toDoubleOrNull()
+                    ),
+                    wind = Wind(
+                        speed = it.wnd?.toDoubleOrNull()
+                    )
+                )
+            )
+        }
+    }
+    return hourlyList
+}
+
+private fun getWeatherText(
+    context: Context,
+    weather: Int?,
+): String? {
+    // TODO: Check with a Mongolian speaker (???) for correct interpretation of the text
```

Comment:
I asked for a proofreader on the Matrix channel. Will merge as is, if no one can help